controllers/btpoperator_controller.go

```diff
@@ -475,7 +480,7 @@ func (r *BtpOperatorReconciler) deleteResources(ctx context.Context, us []*unstr
 	return nil
 }
 
-func (r *BtpOperatorReconciler) reconcileResources(ctx context.Context, s *corev1.Secret) error {
+func (r *BtpOperatorReconciler) reconcileResources(ctx context.Context, baseSecret *corev1.Secret) error {
```

Comment:
How about State of the resource?
Is it in line with what is happening?
Restart could take some time, operator is not operational and we enter `reconcileResources` either from Ready state or from Processing state.